PopPUNK/refine.py

```diff
@@ -454,15 +449,21 @@ def newNetwork2D(y_idx, sample_names, distMat, x_range, y_range, score_idx=0,
     y_max = y_range[y_idx]
     i_vec, j_vec, idx_vec = \
             poppunk_refine.thresholdIterate2D(distMat, x_range, y_max)
-    scores = growNetwork(sample_names,
-                            i_vec,
-                            j_vec,
-                            idx_vec,
-                            x_range,
-                            score_idx,
-                            y_idx,
-                            betweenness_sample,
-                            use_gpu = use_gpu)
+
+    # If everything is in the network, skip this boundary
+    if len(idx_vec) == distMat.shape[0]:
+        scores = [0] * len(x_range)
+    else:
```

Comment:
Do we need an equivalent of this in the 1D refinement?

Reply:
We just give up in this case:
https://github.com/johnlees/PopPUNK/blob/manual_start/PopPUNK/refine.py#L207-L208

Reply:
Should we be consistent, if using the same manual start file for refinement in all 3 possible modes? Or check that the end > greatest distMat entry?

Reply:
I think that in 1D this is a genuine total misspecification of the search range, but in 2D this x can be skipped as there are still plenty of others x values which can work, and sometimes it can be quite difficult to find a range which works for every value

The end > max(distMat) check is a little more fiddly because you have to check the distance to the boundary, so I just went with this simpler one which seemed to show pretty well when something had gone wrong

Reply:
Seems reasonable to me - all looks good, are there changes to `models.py` as well?

Reply:
Yes I forgot to commit those initially, it's just a change to pass the scale so the output with the search range is the same as shown on the plot

Reply:
Although I also changed the indiv refine so the results are stored correctly. I guess that might result in a merge conflict with your branch?